README.adoc

```diff
@@ -2185,6 +2185,28 @@ File.open('testfile') do |f|
 end
 ----
 
+=== Atomic File Operations [[atomic-file-operations]]
+
+When doing file operations after confirming the existence check of a file, frequent parallel file operations may cause problems that are difficult to reproduce.
+Therefore, it is preferable to use atomic file operations.
+
+[source,ruby]
+----
+# bad
+unless File.exist?(path)
+  FileUtils.makedirs(path)
+end
+
+if File.exist?(path)
+  FileUtils.remove(path)
+end
+
+# good
+FileUtils.mkdir_p(path)
+
+FileUtils.rm_rf(path)
```

Comment:
I believe the safe alternative that should be suggested for `FileUtils.remove(path) if File.exist?(path)` should rather be `FileUtils.rm_f(path)` as it doesn't have the same behavior than `FileUtils.rm_rf` on directories

Reply:
Indeed it is. I changed the example. Thank you so much.